typed-racket-lib/typed-racket/base-env/extra-procs.rkt

```diff
@@ -17,7 +17,7 @@
             (quasisyntax/loc stx
               (if (pred val)
                   val
-                  (error (format "Assertion ~a failed on ~v" pred val))))
+                  (error (format "Assertion ~a failed on ~e" pred val))))
```

Comment:
Perhaps use `~e` for the predicate too? Otherwise, the same problem would occur for

```
#lang typed/racket

(struct x ([y : (Listof String)]) #:transparent
  #:property
  prop:procedure
  (λ (si x)
    #f))

(assert 1 (x (make-list 10000 "ouch")))
```

Reply:
Would it be better if we use `(object-name pred)` here?

Reply:
ping @jbclements

Reply:
ping @jbclements